### ISSUE TITLE
From time to time, NGC FAILED: tmpDir Error: ENOTEMPTY: Directory not empty

### ISSUE THREAD

**Issue by mflmartin** [driftyco/ionic-app-scripts#409]
### Short description of the problem:
`ionic run android --device` **SOMETIMES** yields errror, regarding tmpDir

#### What behavior are you expecting?
No error, and good build.

**Steps to reproduce:**
1. ionic run android --device
( Attention! This bug is random and difficult to reproduce)

**Which @ionic/app-scripts version are you using?**
Cordova CLI: 6.4.0
Ionic Framework Version: 2.0.0-rc.2
Ionic CLI Version: 2.1.8
Ionic App Lib Version: 2.1.4
Ionic App Scripts Version: 0.0.39
ios-deploy version: Not installed
ios-sim version: Not installed
OS: Windows 10
Node Version: v6.4.0
Xcode version: Not installed

**Other information:** (e.g. stacktraces, related issues, suggestions how to fix, stackoverflow links, forum links, etc)

**Dirty FIX**: Delete yourself the .tmp folder at your project's root. ionic run (...) etc, again.

[20:37:52]  ngc failed: tmpDir error: Error: ENOTEMPTY: directory not empty, rmdir 
            'D:\Sync\Martin\inProgress\p4q\suntrack_scan\.tmp\pages' 
[20:37:52]  ionic-app-script task: "build" 
[20:37:52]  Error: tmpDir error: Error: ENOTEMPTY: directory not empty, rmdir 
            'D:\Sync\Martin\inProgress\p4q\suntrack_scan\.tmp\pages' 

npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\marti\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
npm ERR! node v6.4.0
npm ERR! npm  v3.10.9
npm ERR! code ELIFECYCLE
npm ERR! suntrack_scan@ build: `ionic-app-scripts build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the suntrack_scan@ build script 'ionic-app-scripts build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the suntrack_scan package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     ionic-app-scripts build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs suntrack_scan
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls suntrack_scan
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\Sync\Martin\inProgress\p4q\suntrack_scan\npm-debug.log

**Comment by Iyashu5040** [comment]
I recall that I had this issue as well. Updating to the latest version ionic app scripts fixed this for me though.

**Comment by mflmartin** [comment]
Which is? How can I check the latest one?

**Comment by dallastjames** [comment]
@mflmartin the latest version is 0.0.44 (npm install @ionic/app-scripts@latest). I've been having that same issue with needing to manually delete the .tmp directory from time to time using the same version of app-scripts that you're running (0.0.39). I haven't upgraded to the latest yet so I can't verify if that will fix your issue.

**Comment by Iyashu5040** [comment]
Please NOTE: there are special steps required when upgrading from below 0.0.42 to above. You can't just npm install latest. See https://github.com/driftyco/ionic-app-scripts/releases for information on releases. And look at https://github.com/driftyco/ionic-app-scripts/releases/tag/v0.0.42 to see the required steps to upgrade.

**Comment by mflmartin** [comment]
Nope, 

I upgraded to those versions... aaaaaaaand, first build: BOOM.

```
ngc failed: tmpDir error: Error: ENOTEMPTY: directory not empty, rmdir 
            'D:\Sync\Martin\inProgress\p4q\suntrack_scan\.tmp\pages' 
[23:37:11]  ionic-app-script task: "build" 
[23:37:11]  Error: tmpDir error: Error: ENOTEMPTY: directory not empty, rmdir 
            'D:\Sync\Martin\inProgress\p4q\suntrack_scan\.tmp\pages' 

```

:( 


 ordova CLI: 6.4.0
Ionic Framework Version: 2.0.0-rc.2
Ionic CLI Version: 2.1.8
Ionic App Lib Version: 2.1.4
Ionic App Scripts Version: 0.0.44
ios-deploy version: Not installed
ios-sim version: Not installed
OS: Windows 10
Node Version: v6.4.0
Xcode version: Not installed

**Comment by danbucholtz** [comment]
Hi all,

This is resolved in `0.0.48` which was just published.

Please let me know if you have any issues.

Thanks,
Dan
